core/src/main/java/org/apache/iceberg/rest/auth/OAuth2Util.java

```diff
@@ -550,31 +512,29 @@ public static AuthSession fromAccessToken(
         RESTClient client,
         ScheduledExecutorService executor,
         String token,
-        Long defaultExpiresInMillis,
+        Long defaultExpiresAtMillis,
         AuthSession parent) {
       AuthSession session =
           new AuthSession(
               parent.headers(), token, OAuth2Properties.ACCESS_TOKEN_TYPE, parent.credential());
 
-      long startTimeMillis = System.currentTimeMillis();
-      Long expiresAtMillis = session.expiresAtMillis();
+      Long expiresAtMillis = null;
 
-      if (isExpired(expiresAtMillis, startTimeMillis)) {
+      if (session.expiresAtMillis() != null
+          && session.expiresAtMillis() <= System.currentTimeMillis()) {
```

Comment:
Why did you change the use of `expiresAtMillis`? Now there are 4 calls to `session.expiresAtMillis()` rather than using a variable.